shared/src/commonMain/kotlin/io/customer/shared/di/KMMStaticComponent.kt

```diff
@@ -0,0 +1,33 @@
+package io.customer.shared.di
+
+import io.customer.shared.sdk.config.BuildConfigurations
+import io.customer.shared.sdk.config.getBuildConfigurations
+import io.customer.shared.util.*
+
+/**
+ * Static component dependency graph to satisfy independent dependencies from single place. All
+ * other graphs should never redefine dependencies available here unless extremely necessary. The
+ * structure of DIGraphs would something like:
+ *
+ * -> StaticGraph (Independent)
+ * --> PlatformGraph (Depends on StaticGraph and Platform)
+ * ---> WorkspaceGraph (Depends on StaticGraph, PlatformGraph)
+ *
+ * The class should only contain dependencies matching the following criteria:
+ * - dependencies that may be required without SDK initialization.
+ * - dependencies that are lightweight and are not dependent on SDK initialization.
+ */
```

Comment:
This might be helpful in understanding the idea behind 3 levels of graphs.

Reply:
This idea defined in the comment of using different levels of DiGraphs seems to try and solve a problem of, "What do we do when the SDK isn't initialized?."

I think this solution is a bit overkill for trying to solve this problem. I have a good feeling we could come up with something more simple. 

This is difficult to wrap my head around already. I can only imagine maintaining it will be more difficult.

Reply:
Can you share some idea on how can we solve this?

We have faced several issues when working on wrapper SDKs and having multiple graphs has helped us in this. I'm happy to adapt any solution that solves those challenges in a simpler way but having multiple levels has helped us in Android. 

Also, maintaining this won't be difficult if we can define some simple rules for adding classes to this graph. Can you specify what makes you think it is difficult to maintain?